.github/workflows/upload-android-test-specs.yml

```diff
@@ -45,50 +29,3 @@ jobs:
       models: stories110M
       devices: samsung_galaxy_s22
       delegates: xnnpack
-      test_spec: https://gha-artifacts.s3.amazonaws.com/${{ github.repository }}/${{ github.run_id }}/artifacts/android-llm-device-farm-test-spec.yml
-
-  upload-android-test-spec:
```

Comment:
It seems the upload step is moved to android-perf.yml right after the test-spec is specialized and instantiated, w/o validation. Given that, do we still need to keep a validation step here? How would it catch and prevent bad test-spec being uploaded and used?

Reply:
In this new setup, the specialized test spec will just be an artifact of the current job uploaded at `https://gha-artifacts.s3.amazonaws.com/${{ github.repository }}/${{ github.run_id }}/artifacts/${{ matrix.model }}_${{ matrix.delegate }}/android-llm-device-farm-test-spec.yml`.  If the spec is bad, the subsequent benchmark job would fail.

The validation workflow here now acts just as a trigger to call android-perf workflow when the spec template changes.

Finally, the upload step here is being deleted (not moving to android-perf) because there would not be a single share spec file at `s3://ossci-android/executorch/android-llm-device-farm-test-spec.yml` anymore.  It couldn't be because each specialized spec now has different model+delegation.

Reply:
Maybe this can be simplified and merged into `android-perf.yml` by triggering a run when the test spec template is modified?

Reply:
We could do that but the job would need to use the default value in CRON_DEFAULT_MODELS, it seems overkill to me to test them all just to validate the spec.  But I think I could tweak it a bit to use a different value for pull_request and keep the full list for scheduled jobs.

Reply:
Here it the example run https://github.com/pytorch/executorch/actions/runs/12150009281 without `upload-android-test-specs.yml`

Reply:
> job would need to use the default value in CRON_DEFAULT_MODELS, it seems overkill to me to test them all just to validate the spec.

I see. This is a fair point. Maybe we could add new defaults or change the defaults for test-spec validation to `android-perf.yml`? 
I don't have strong option to merge it to the `android-perf.yml`, thinking in that direction because job `upload-android-test-specs.yml` now is essentially validating the `android-perf.yml`, the specialized test-spec generated on the fly is already part of it.